Bad interactions in JSDOMNodeJSEnv error handling on JSDom 10-12

I can't get a run of utest 0.6.4 or even 0.6.5 with the opcode change, of scalajs-react tests to complete with SJS 0.6.25. I constantly one of the following errors which kills all subsequent tests. I've had to downgrade SJS to 0.6.24

```
org.scalajs.testcommon.RPCCore$ClosedException
	at org.scalajs.testcommon.RPCCore.helpClose(RPCCore.scala:193)
	at org.scalajs.testcommon.RPCCore.close(RPCCore.scala:185)
	at org.scalajs.testadapter.ComJSEnvRPC.org$scalajs$testadapter$ComJSEnvRPC$$super$close(ComJSEnvRPC.scala:50)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:50)
Caused by: org.scalajs.jsenv.ComJSEnv$ComClosedException: JSCom has been closed
	at org.scalajs.jsenv.nodejs.AbstractNodeJSEnv$NodeComJSRunner$class.receive(AbstractNodeJSEnv.scala:284)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.jsenv.ComJSRunner$class.receive(ComJSRunner.scala:15)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:31)
Caused by: java.io.EOFException
	at java.io.DataInputStream.readInt(DataInputStream.java:392)
	at org.scalajs.jsenv.nodejs.AbstractNodeJSEnv$NodeComJSRunner$class.receive(AbstractNodeJSEnv.scala:274)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.jsenv.ComJSRunner$class.receive(ComJSRunner.scala:15)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:31)
```

or

```
org.scalajs.testcommon.RPCCore$ClosedException
	at org.scalajs.testcommon.RPCCore.helpClose(RPCCore.scala:193)
	at org.scalajs.testcommon.RPCCore.close(RPCCore.scala:185)
	at org.scalajs.testadapter.ComJSEnvRPC.org$scalajs$testadapter$ComJSEnvRPC$$super$close(ComJSEnvRPC.scala:50)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:50)
Caused by: org.scalajs.jsenv.ComJSEnv$ComClosedException: JSCom has been closed
	at org.scalajs.jsenv.nodejs.AbstractNodeJSEnv$NodeComJSRunner$class.receive(AbstractNodeJSEnv.scala:284)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.jsenv.ComJSRunner$class.receive(ComJSRunner.scala:15)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:31)
Caused by: java.io.EOFException
	at java.io.DataInputStream.readInt(DataInputStream.java:392)
	at org.scalajs.jsenv.nodejs.AbstractNodeJSEnv$NodeComJSRunner$class.receive(AbstractNodeJSEnv.scala:274)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.jsenv.ComJSRunner$class.receive(ComJSRunner.scala:15)
	at org.scalajs.jsenv.nodejs.JSDOMNodeJSEnv$ComDOMNodeRunner.receive(JSDOMNodeJSEnv.scala:62)
	at org.scalajs.testadapter.ComJSEnvRPC$$anon$1.run(ComJSEnvRPC.scala:31)
```

--- a/js-envs/src/main/scala/org/scalajs/jsenv/nodejs/JSDOMNodeJSEnv.scala
+++ b/js-envs/src/main/scala/org/scalajs/jsenv/nodejs/JSDOMNodeJSEnv.scala
@@ -84,6 +84,15 @@ class JSDOMNodeJSEnv private[jsenv] (
            |    var virtualConsole = new jsdom.VirtualConsole()
            |      .sendTo(console, { omitJSDOMErrors: true });
            |    virtualConsole.on("jsdomError", function (error) {
+           |      /* #3458 Counter-hack the hack that React's development mode
+           |       * uses to bypass browsers' debugging tools. If we detect
+           |       * that we are called from that hack, we do nothing.
+           |       */
+           |      var isWithinReactsInvokeGuardedCallbackDevHack_issue3458 =
+           |        new Error("").stack.indexOf("invokeGuardedCallbackDev") >= 0;
+           |      if (isWithinReactsInvokeGuardedCallbackDevHack_issue3458)
+           |        return;
+           |
            |      try {
            |        // Display as much info about the error as possible
            |        if (error.detail && error.detail.stack) {

